pkg/storage/inmemory/article.go

```diff
@@ -0,0 +1,97 @@
+package inmemory
+
+import (
+	"errors"
+	"github.com/reiver/go-porterstemmer"
+	"news-aggregator/pkg/model"
+	"strings"
+	"time"
+)
+
+// ArticleInMemory is a struct that contains the in-memory database for articles.
+type ArticleInMemory struct {
+	Articles []model.Article
+	nextID   int
+}
+
+func New() *ArticleInMemory {
+	return &ArticleInMemory{
+		Articles: []model.Article{},
+		nextID:   1,
+	}
+}
+
+// GetAll returns all articles in the database.
+func (a *ArticleInMemory) GetAll() ([]model.Article, error) {
+	return a.Articles, nil
+}
+
+// Create adds a new article to the database.
+func (a *ArticleInMemory) Create(article model.Article) (model.Article, error) {
+	article.Id = a.nextID
+	a.nextID++
+	a.Articles = append(a.Articles, article)
+	return article, nil
+}
+
+// Delete removes the article with the given ID from the database.
+func (a *ArticleInMemory) Delete(id int) error {
+	for i, article := range a.Articles {
+		if article.Id == id {
+			a.Articles = append(a.Articles[:i], a.Articles[i+1:]...)
+			return nil
+		}
+	}
+	return errors.New("article not found")
```

Comment:
instead of this, you can create your custom error type 
https://www.digitalocean.com/community/tutorials/creating-custom-errors-in-go

Reply:
I've read about this, but as I understood, it just wraps an error but inside it calls just the same errors.New. Honestly, in this case, I think there is no need to create a custom type of error